packages/azure-openapi-validator/core/src/runner.ts

```diff
@@ -48,20 +50,32 @@ export class LintRunner<T> {
       const targetDefinition = openapiDefinition
       for (const given of givens) {
         for (const section of nodes(targetDefinition, given)) {
-          const fiieldMatch = rule.then.fieldMatch
-          if (fiieldMatch) {
-            for (const subSection of nodes(section.value, fiieldMatch)) {
-              const location = section.path.slice(1).concat(subSection.path.slice(1))
-              const args = getArgs(rule, subSection.value, targetDefinition, location)
+          const fieldMatch = rule.then.fieldMatch
+          if (fieldMatch) {
+            for (const subSection of nodes(section.value, fieldMatch)) {
+              const location: string[] = section.path.slice(1).concat(subSection.path.slice(1))
+              await processRule(rule, subSection.value, location)
+            }
+          } else {
+            const location: string[] = section.path.slice(1)
+            await processRule(rule, section.value, location)
+          }
+
+          async function processRule(rule: IRule<any>, section: any, location: string[]) {
```

Comment:
FYI: the code that is in `processRule` was duplicated and did not had the `try/catch`. Hence I first deduplicated it into this function before adding the `try/catch`.